tests/type-constraints-tests.egg

```diff
@@ -0,0 +1,10 @@
+(datatype Operand)
+(sort VecOperandBase (Vec Operand))
+(datatype VecOperand (VO VecOperandBase))
```

Comment:
Sorry I think I missed it maybe in an earlier PR, what does the `V0` do?

Reply:
Ah it's a wrapper trick we used in eggcc that allows us to union two vectors. It's very specific to our application.

Reply:
Ah sorry, I was just misreading the datatype constructor as a sort constructor and thought that `V0` was a presort. Nevermind!